internal/expr/parser.go

```diff
@@ -77,17 +78,32 @@ type expression interface {
 	expr()
 }
 
-// valueAccessor stores information for accessing a Go value. It consists of a
-// type name and some value within it to be accessed. For example: a field of a
-// struct, or a key of a map.
-type valueAccessor struct {
+// valueAccessor stores information about how to access a Go value. For example:
+// by index, by key, or by slice syntax.
+type valueAccessor interface {
+	getTypeName() string
```

Comment:
It is only prefixed by `get` because it clashes with the `typeName` field of `memberAccessor`. The problem is it is difficult to come up with a better name for the field and/or the method.